coil-video/src/main/java/coil/decode/VideoFrameDecoder.kt

```diff
@@ -0,0 +1,54 @@
+@file:Suppress("unused")
+
+package coil.decode
+
+import android.content.Context
+import android.media.MediaMetadataRetriever
+import coil.bitmap.BitmapPool
+import coil.size.Size
+import okio.BufferedSource
+import okio.sink
+import java.io.File
+
+/**
+ * A [Decoder] that uses [MediaMetadataRetriever] to fetch and decode a frame from a video.
+ *
+ * NOTE: [VideoFrameDecoder] creates a temporary copy of the video on the file system. This may cause the decode
+ * process to fail if the video being decoded is very large and/or the device is very low on disk space.
+ */
+class VideoFrameDecoder(private val context: Context) : Decoder {
+
+    private val delegate = VideoFrameDecoderDelegate(context)
+
+    override fun handles(source: BufferedSource, mimeType: String?): Boolean {
+        return mimeType != null && mimeType.startsWith("video/")
+    }
+
+    override suspend fun decode(
+        pool: BitmapPool,
+        source: BufferedSource,
+        size: Size,
+        options: Options
+    ): DecodeResult {
+        val tempFile = File.createTempFile("tmp", null, context.cacheDir.apply { mkdirs() })
```

Comment:
We don't need write permissions to do this, right?

Reply:
Yep! The `cacheDir` is free to write to.